StickyHeader | Review comments incorporated

Vipul6 opened the pull request.

Commit:
issue-111 | With Scripts hoc created

Commit:
issue-111 | With Scripts hoc created

Commit:
issue-72 | React hooks based hoc for script loading

Commit:
Accordion component

Commit:
Merge branch 'feature/design-system-basics' of https://github.com/pagesource/atomic-react-components into feature/accordion

Commit:
Accordion development in progress

Commit:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/accordion

Commit:
Accordion changes

Commit:
Accordion changes

Commit:
test commit

Commit:
added new component

Commit:
added pagination component

Commit:
Accordion lint fixes

Commit:
Accordion fixes'

Commit:
added active class when item is active

Commit:
updated the functionality

Commit:
Accordion - Review comments incorporated

Commit:
code changes

Commit:
Accordion changes

Commit:
Accordion changes

Commit:
Accordion fixes

Commit:
Accordion fixes

salilkapoor committed:
Initial Commit.

Commit:
Accordion - Storybook changes done, test cases added, accessibility changes done

Commit:
Accordion changes

Commit:
Accordion changes

Commit:
Accordion

Commit:
Accordion changes

Commit:
Accordion changes

Commit:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/accordion

Commit:
Pull from develop branch

Commit:
Implemented Carousel

1. Equipped with 4 variations (Legacy, Center Mode, Thumbnails,
Responsive)
2. Can be handled with props easily.
3. Provided separate prop for button variations and other props.

salilkapoor committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/cookie-disclaimer

# Conflicts:
#	.jest-test-results.json
#	yarn.lock

salilkapoor committed:
Cookie Disclaimer: Styled Components Checked in

Refined test cases according to styles.

salilkapoor committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/cookie-disclaimer

# Conflicts:
#	.jest-test-results.json

salilkapoor committed:
Cookie Disclaimer: Styled Components Checked in

Refined test cases according to styles.

salilkapoor committed:
Cookie Disclaimer: Styled Components Checked in

Refined test cases according to styles.

Commit:
Accordion changes'

Commit:
Updated test cases'

salilkapoor committed:
Cookie Disclaimer: Styled Components Checked in

Refined test cases according to styles.

Commit:
Accordion review comments incorporated

Commit:
Accordion changes

Commit:
updated my local branch

salilkapoor committed:
Cookie Disclaimer (WIP): Styled Components Checked in

Added functional behaviour to the component.

salilkapoor committed:
Cookie Disclaimer (WIP): Styled Components Checked in

Added functional behaviour to the component.

Commit:
refactored the pagination component

Commit:
Review comments implemented

salilkapoor committed:
Merge branch 'feature/cookie-disclaimer' of https://github.com/salilkapoor/atomic-react-components into feature/cookie-disclaimer

# Conflicts:
#	.jest-test-results.json
#	lib/styles/themes/base/tokens/colors.css

salilkapoor committed:
Test permission issue.

Commit:
fixed the demo page

vinodloha commented on utils/util.js:
```diff
@@ -0,0 +1,7 @@
+const useHandleScroll = (elemRef, setterFunc) => {
```
The name of the hook is ambiguous and also not placed correctly. We are keeping hooks at lib/hooks/*

Vipul6 replied:
It has been taken care.

vinodloha commented on lib/index.js:
```diff
@@ -79,4 +80,6 @@ export {
   TypeaheadVanilla,
   Video,
   VideoVanilla,
+  StickyHeader,
```
Follow the alphabetical order

Vipul6 replied:
It has been taken care.

vinodloha commented on lib/components/molecules/StickyHeader/StickyHeader.js:
```diff
@@ -0,0 +1,31 @@
+
+  return (
+    <div aria-label="Sticky Header" className={`${isSticky ? className : ''}`} ref={wrapperRef}>
+      <header>{children}</header>
```
@Vipul6 I don't think you have fixed the original problem i had highlighted. This component is causing a flicker when it gets sticky and also hiding the page content below itself.

Vipul6 replied:
Fixed.

vinodloha reviewed (changes requested):
This still needs few fixes

salilkapoor committed:
Cookie Disclaimer (WIP): Styled Components Checked in

Added functional behaviour to the component.

salilkapoor committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/cookie-disclaimer

# Conflicts:
#	.jest-test-results.json
#	yarn.lock

salilkapoor committed:
Hotfix (Icon Config): Fixed the icon component

Commit:
Hero Banner Component

Commit:
Accordion review comments incorporated

Commit:
Accordion changes

Commit:
Remaining review comments implemented

Commit:
Accordion changes

Commit:
Merge branch 'develop' of github.com:pagesource/atomic-react-components into feature/Sticky-Header-Develop

Commit:
Merge branch 'feature/Sticky-Header-Develop' of github.com:pagesource/atomic-react-components into feature/Sticky-Header-Develop

Commit:
Review comments fixed

Commit:
Merge branch 'develop' of github.com:pagesource/atomic-react-components into feature/Sticky-Header-Develop

Commit:
Merge branch 'develop' into bugfix/icon-config

vinodloha committed:
Merge pull request #200 from pagesource/bugfix/icon-config

Hotfix (Icon Config): Fixed the icon component

vinodloha commented on lib/components/molecules/StickyHeader/StickyHeader.js:
```diff
@@ -0,0 +1,36 @@
+
+  return (
+    <Wrapper
+      aria-label="Sticky Header"
```
Don't hard code any string in the component, take them through props please.

aashutosh-prakash replied:
aria-label wont work on non-interactive elements like Wrapper (originally a div tag).
Removed it from code.

Commit:
Merge branch 'develop' into feature/cookie-disclaimer

vinodloha commented on lib/components/molecules/StickyHeader/StickyHeader.story.js:
```diff
@@ -0,0 +1,40 @@
+          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has
+          been the industrys
+        </p>
+        <StickyHeader>
```
Tested this, still not working as expected. When this instance gets sticky it should not overlap the above header unless explicitly set to do so. You had to create a prop that takes sticky position value. 

We had discussed this in quite a detail.

vinodloha commented on lib/components/molecules/StickyHeader/StickyHeader.js:
```diff
@@ -0,0 +1,36 @@
+      ref={wrapperRef}
+      elmRef={wrapperRef}
+    >
+      <header>{children}</header>
```
If there could be multiple Header tag on page, it should be wrapped in Section

vinodloha commented on lib/components/molecules/StickyHeader/StickyHeader.style.js:
```diff
@@ -0,0 +1,16 @@
+
+export const Wrapper = styled.div`
+  position: relative;
+  height: ${props => props.elmRef.current != null && `${props.elmRef.current.offsetHeight}px`};
```
This should be set inline on the element.

vinodloha commented on lib/hooks/useScroll.js:
```diff
@@ -0,0 +1,7 @@
+const useScroll = (elemRef, setterFunc) => {
```
It should be throttled/debounced, else it will execute on each pixel

vinodloha reviewed (changes requested):
Needs more improvement.

SakshiGrg committed:
Toast-Icon-Component-Dependency-Resolved

vinodloha committed:
Merge pull request #202 from pagesource/feature/Toast-Component-IconDependency-190

Toast-Icon-Component-Dependency-Resolved

Commit:
modified pagination

Commit:
added test cases

salilkapoor committed:
Cookie Disclaimer (WIP): Styled Components Checked in

Added functional behaviour to the component.

Commit:
jest test result

Commit:
taken latest pull from develop

Commit:
fixed some issues

Commit:
Accordion changes

salilkapoor committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/cookie-disclaimer

# Conflicts:
#	yarn.lock

vinodloha committed:
Merge pull request #199 from pagesource/feature/cookie-disclaimer

Cookie Disclaimer: Feature disclaimer code pushed for initial review

Commit:
Skeleton initial commit'

Commit:
removed unused code

Commit:
updated with latest code

Commit:
Skeleton changes

Commit:
Skeleton changes

Commit:
Hero Banner Component

Commit:
Skeleton changes

Commit:
Fixed review Comments on Hero Banner

Commit:
Fixed review Comments on Hero Banner

vinodloha committed:
Merge pull request #201 from pagesource/feature/hero-component-89

Hero Banner Component

Commit:
Skeleton review comments incorporated

Commit:
incorporated review comments

Commit:
updated teh code base

Commit:
correction in aria also removed extra codes

Commit:
Exported the components out of Library

Commit:
Updated the snapshots

vinodloha committed:
Merge pull request #169 from pagesource/feature/molecule-pagination

Feature/molecule pagination

Commit:
Accordion changes

Commit:
Accordion

Commit:
Exported the components out of Library, Fixed labels, types and Typos

Commit:
Fixed Storybook Hiererchy

Commit:
Merge branch 'develop' into feature/accordion

vinodloha committed:
Merge pull request #168 from pagesource/feature/accordion

Feature/accordion

Commit:
Skeleton changes

vinodloha committed:
Merge pull request #192 from pagesource/feature/atomic-carousel

Implemented Carousel with Declarative Approach

Commit:
Skeleton changes

SakshiGrg committed:
ProgressIndicator Component Review Required

SakshiGrg committed:
Progress indicator Review

ritugupta64 committed:
fix/hoc will consume a component and script

1) added mock data
2) added story to view the component in web

SakshiGrg committed:
HTML5 Progress implemented

Commit:
Skeleton review comments

vinodloha committed:
Merge pull request #206 from pagesource/feature/skeleton

Skeleton initial commit

SakshiGrg committed:
FEAT/J
TEST COMMIT

SakshiGrg committed:
feat/(jira-123)

changed kjdsfihf

close #123

SakshiGrg committed:
feat/(jira-123) added test file

added test file to test git commit message

issue#jira-123 closed

SakshiGrg committed:
feat/(issue-179) added test file

added to test git commit message

Commit:
issue-11 With scripts review comments fixed

Commit:
issue-11 With scripts review comments fixed: test cases added

vinodloha committed:
Merge pull request #209 from pagesource/feature/ProgressIndicator-ProgressSemantic-179

Feature/progress indicator progress semantic 179

vinodloha committed:
Merge pull request #115 from pagesource/issue/issue-111-withScripts

 issue-111 | With Scripts hoc created for custom j.s loading

anilkumar23Sap committed:
converted popover class component to stateless component with hooks

anilkumar23Sap committed:
converted tabs class component to stateless component with hooks

Commit:
feat(add-commitlint): add commitlint

add rule in husky for local
add rule in travis.yml for ci/cd

Commit:
fix: upgrade packages to latest version

Commit:
fix: linting issuesm test cases snapshots

Commit:
fix breaking change: shell option in execa

vinodloha committed:
Merge pull request #219 from pagesource/fix/update-packages

Fix/update packages

Commit:
feat(test-lint): test lint

test lint

Commit:
feat(add-connit-lint): update commitlint

add commit lint rul in commitlintrc.js

Commit:
feat(test-lint): test lint

test lint

Commit:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into commitlintNew

Commit:
feat(test-lint): test lint

test lint

Commit:
feat(test-lint): test lint

fix lint

Commit:
fix(test-lint): fix lint

fix lint

Commit:
feat(test-lint): test lint

test lint

Commit:
feat(test-lint): test lint

remove snapshot update

Commit:
feat(test-lint): test lint

test lint

Commit:
feat(fix-lint): fix lint

fix lint

vinodloha committed:
Merge pull request #220 from ankush29/commitlintNew

Commitlint new

vinodloha committed:
Merge pull request #216 from pagesource/feature/class-component-to-hooks

Feature/class component to hooks

Commit:
fix: flow issues

vinodloha committed:
Merge pull request #215 from pagesource/fix/flow-issues

fix: flow issues fix

Commit:
fix: added flow check in npm validate

Commit:
fix: added flow as pre push hook

Commit:
fix: added flow check in pre commit hook

vinodloha committed:
Merge pull request #223 from pagesource/fix/add-flow-check

fix: added flow check in npm validate

imsauravsingh committed:
fix: fixed pr comment issue

fixed pr comment issue

imsauravsingh committed:
fix: fixed snapshot issue

fixed snapshot issue

vinodloha committed:
Merge pull request #227 from pagesource/feature/global-search-box-commitlint

Feature/global search box commitlint

Commit:
Merge branch 'feature/Sticky-Header-Develop' of https://github.com/pagesource/atomic-react-components into feature/Sticky-Header-Develop

Commit:
feat: tag, husky config, reviews, glitches
1. add tag prop for adding different tags
2. updated husky config
3. added feature defaultTopHeight
4. refactor css width approach
5. remove extra code
6. added reviews suggestion

aashutosh-prakash commented:
Raised new PR with commit lint fixes.
